flow/raster_cacheable_entry.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/raster_cacheable_entry.h"
+
+#include <utility>
+
+#include "flutter/display_list/display_list.h"
+#include "flutter/flow/layers/cacheable_layer.h"
+#include "flutter/flow/layers/layer.h"
+#include "flutter/flow/raster_cache.h"
+#include "flutter/flow/raster_cache_key.h"
+#include "include/core/SkMatrix.h"
+
+namespace flutter {
+
+void CacheableLayerWrapper::TryToPrepareRasterCache(PrerollContext* context,
+                                                    const SkMatrix& matrix) {
+  layer_->TryToPrepareRasterCache(context, matrix);
+}
+
+void CacheableDisplayListWrapper::TryToPrepareRasterCache(
+    PrerollContext* context,
+    const SkMatrix& matrix) {
+  if (auto* cache = context->raster_cache) {
+    SkRect bounds =
+        display_list_->bounds().makeOffset(offset_.x(), offset_.y());
+    TRACE_EVENT0("flutter", "DisplayListLayer::RasterCache");
+    if (context->cull_rect.intersects(bounds)) {
+      cache->Prepare(context, display_list_, matrix, offset_);
+    } else {
+      // Don't evict raster cache entry during partial repaint
+      cache->Touch(display_list_, matrix);
+    }
+  }
+}
+
+void CacheableSkPictureWrapper::TryToPrepareRasterCache(
+    PrerollContext* context,
+    const SkMatrix& matrix) {
+  if (auto* cache = context->raster_cache) {
+    SkRect bounds =
+        sk_picture_->cullRect().makeOffset(offset_.x(), offset_.y());
+
+    TRACE_EVENT0("flutter", "PictureLayer::RasterCache (Preroll)");
+    if (context->cull_rect.intersects(bounds)) {
+      if (cache->Prepare(context, sk_picture_, matrix, offset_)) {
+        context->subtree_can_inherit_opacity = true;
```

Comment:
This is useless. This flag in PrerollContext is only examined during the calls to Preroll. Setting it here during the cache evaluation phase does nothing as there is no parent to evaluate the flag any more. This method is called directly by code that ignores the values left in the PrerollContext.